crates/matrix-sdk-base/src/client.rs

```diff
@@ -624,6 +676,7 @@ impl BaseClient {
 
             room_info.update_summary(&new_info.summary);
             room_info.set_prev_batch(new_info.timeline.prev_batch.as_deref());
+            room_info.mark_state_fully_synced();
```

Comment:
This isn't correct, last review round we mentioned that you will be missing members here if lazy loading is enabled, which we assume.

Reply:
This is purely about room state, not member events, as far as I understand. 
The members_synced and room state synced flags are separate.